test/controllers/tasks_controller_test.rb

```diff
@@ -1,126 +1,285 @@
-
+      
       # Assert
       must_respond_with :redirect
-      expect(flash[:error]).must_equal "Could not find task with id: -1"
+      expect(flash[:error]).must_equal 'Could not find task with id: -1'
     end
   end
-
-  describe "new" do
-    it "can get the new task page" do
-      skip
-
+  
+  describe 'new' do
+    it 'can get the new task page' do
+      # skip
+      
       # Act
       get new_task_path
-
+      
       # Assert
       must_respond_with :success
     end
   end
-
-  describe "create" do
-    it "can create a new task" do
-      skip
-
+  
+  describe 'create' do
+    it 'can create a new task' do
+      # skip
+      
       # Arrange
       task_hash = {
         task: {
-          name: "new task",
-          description: "new task description",
-          completion_date: nil,
-        },
+          id: 1,
+          name: 'new task',
+          description: 'new task description',
+          completed_at: nil,
+        }
       }
-
+      
       # Act-Assert
       expect {
         post tasks_path, params: task_hash
-      }.must_change "Task.count", 1
-
+      }.must_change 'Task.count', 1
+      
       new_task = Task.find_by(name: task_hash[:task][:name])
       expect(new_task.description).must_equal task_hash[:task][:description]
-      expect(new_task.due_date.to_time.to_i).must_equal task_hash[:task][:due_date].to_i
-      expect(new_task.completed).must_equal task_hash[:task][:completed]
-
+      expect(new_task.completed_at).must_equal task_hash[:task][:completed_at]
+      
       must_respond_with :redirect
       must_redirect_to task_path(new_task.id)
     end
   end
-
-  # Unskip and complete these tests for Wave 3
-  describe "edit" do
-    it "can get the edit page for an existing task" do
-      skip
-      # Your code here
+  
+  # Wave 3
+  describe '#edit' do
+    describe 'when the task exists' do
+      it 'returns 200' do 
+        # Setup
+        task = Task.create(
+          id: 1, 
+          name: 'sample task', 
+          description: 'this is an example for a test',
+          completed_at: Time.now + 5.days
+        )
+        
+        # Act
+        get edit_task_path(task)
+        
+        # Assert
+        must_respond_with :success
+      end
     end
-
-    it "will respond with redirect when attempting to edit a nonexistant task" do
-      skip
-      # Your code here
+    
+    describe 'when the task does not exist' do
+      it 'redirects the user to the tasks list' do
+        # Act
+        get edit_task_path(-1)
+        # Assert
+        must_respond_with :redirect
+        must_redirect_to tasks_path
+      end
     end
   end
-
-  # Uncomment and complete these tests for Wave 3
-  describe "update" do
-    # Note:  If there was a way to fail to save the changes to a task, that would be a great
-    #        thing to test.
-    it "can update an existing task" do
-      # Your code here
+  
+  describe '#update' do
+    describe 'when the task exists' do
```

Comment:
Your describe and it arguments are a bit redundant. Consider whether you need both of them.